### PR TITLE
feat: troubleshooting page search

### PR DESCRIPTION
**Pull request by phvm**
#### What is the purpose of this pull request?

Implement a search bar for the troubleshooting page

#### What problem is this solving?

Other cards list pages have a search bar except the troubleshooting one

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: src/components/troubleshooting-card/index.tsx [hunk 1]
```diff
@@ -9,23 +9,17 @@ import type { TroubleshootingDataElement } from 'utils/typings/types'
 export default function TroubleshootingCard({
   title,
   slug,
-  tags,
+  productTeam,
```
Comment (PedroAntunesCosta) [inline src/components/troubleshooting-card/index.tsx:12]: This is really suposed to be `tags`. In the current help-center-content state we must still run the script that puts them into the frontmatter. Then they'll be rendered.

File: src/pages/troubleshooting/index.tsx [hunk 2]
```diff
@@ -48,15 +48,16 @@ const TroubleshootingPage: NextPage<Props> = ({
...
-      const hasSearch: boolean = troubleshoot.title
-        .toLowerCase()
-        .includes(search.toLowerCase())
-      return hasSearch && hasFilters
-    })
+    const data = troubleshootingData
+      .filter((troubleshoot) => troubleshoot.status === 'PUBLISHED')
+      .filter((troubleshoot) => {
+        const hasFilters: boolean =
+          filters.length === 0 || filters.includes(troubleshoot.productTeam)
```
Comment (PedroAntunesCosta) [inline src/pages/troubleshooting/index.tsx:55]: This is suposed to work with the tags.

File: src/pages/troubleshooting/index.tsx [hunk 3]
```diff
@@ -220,7 +221,7 @@ export async function getStaticProps({
               slug: data.slug,
               createdAt: String(frontmatter.createdAt),
               updatedAt: String(frontmatter.updatedAt),
-              tags: String(frontmatter.tags ?? '').split(','),
+              productTeam: String(frontmatter.productTeam),
```
Comment (PedroAntunesCosta) [inline src/pages/troubleshooting/index.tsx:224]: Tags here too.

File: src/utils/constants.ts [hunk 4]
```diff
@@ -581,13 +581,58 @@ export const TroubleshootingFilters = (intl: IntlShape) => {
...
     name: intl.formatMessage({ id: 'troubleshooting_filter_module.title' }),
     options: [
-      { id: 'Catalog', name: 'Catalog' },
-      { id: 'Order', name: 'Order' },
-      { id: 'Shipping', name: 'Shipping' },
-      { id: 'VTEX Admin', name: 'VTEX Admin' },
-      { id: 'Prices', name: 'Prices' },
-      { id: 'Message Center', name: 'Message Center' },
-      { id: 'Trade Policy', name: 'Trade Policy' },
+      {
```
Comment (PedroAntunesCosta) [inline src/utils/constants.ts:584]: I am going to try and generate a list with the script that rearranges the tags. But for now we can use a manual approximation, even if we don't check every article.

File: src/utils/typings/types.ts [hunk 5]
```diff
@@ -106,7 +106,7 @@ export type FaqCardDataElement = {
 export type TroubleshootingDataElement = {
   title: string
   slug: string
-  tags: string[]
+  productTeam: string
```
Comment (PedroAntunesCosta) [inline src/utils/typings/types.ts:109]: Tags here too.

**Review by PedroAntunesCosta — CHANGES_REQUESTED** [review]
I am going too fix the tags in the content repo. Then we'll be able to test this properly.

**Comment by julia-rabello** [comment]
I reverted the commits that replaced tags with product team and the tags seem to be working properly now.

image

However, the filters are influenced by the values in constants.ts. Since they're not really constants (they depend on the content of markdown files), we need to get these values from help-center-content troubleshooting files. I'm not sure how to generate that list dynamically, and add localization as well (tags are different depending on the language). Do you have any idea how to do that? @PedroAntunesCosta 

Should we remove filters while this is not ready? Or just have a static list based on troubleshooting guides so far?

**Review by PedroAntunesCosta — CHANGES_REQUESTED** [review]
It works! But I think we should create a static list for the filters. I have some more considerations on this:
- Would all tags be too many for the filter picking UI?
- If so, should we list only products? I feel like this is not the way knowledge should be organized in troubleshooting, but it might help.

**Review by PedroAntunesCosta — APPROVED** [review]

**Pull request merged**